src/Carousel.js

```diff
@@ -126,315 +134,350 @@ const defaultProps = {
   touch: true,
 };
 
-class Carousel extends React.Component {
-  state = {
+const Carousel = React.forwardRef((uncontrolledProps, ref) => {
+  let {
+    activeIndex: activeIndexProp,
+    as: Component = 'div',
+    bsPrefix,
+    children,
+    className,
+    controls,
+    fade,
+    indicators,
+    interval,
+    keyboard,
+    nextIcon,
+    nextLabel,
+    onSelect,
+    onSlideEnd,
+    pauseOnHover,
+    prevIcon,
+    prevLabel,
+    slide,
+    touch,
+    wrap,
+    ...props
+  } = useUncontrolled(uncontrolledProps, {
+    activeIndex: 'onSelect',
+  });
+
+  const prefix = useBootstrapPrefix(bsPrefix, 'carousel');
+
+  const [classPair, setClassPair] = useState({
     prevClasses: '',
     currentClasses: 'active',
-    touchStartX: 0,
-  };
-
-  isUnmounted = false;
+  });
+  let activeIndex = useRef(undefined);
+  let previousActiveIndex = useRef(undefined);
+  let direction = useRef('');
+
+  const isUnmounted = useRef(true);
```

Comment:
use `@restart/hooks/useMounted`

Reply:
thank you for the prompt review! :) 
unfortunately I wasn't able to use `useMounted` because it works in a bit different way, I tried including it, but it caused a lot of tests to fail. 
I checked the `useMounted` source code and it seems to be setting `mounted = true` initially. I am not quite if it was a bug or intentional.

Reply:
eh? why shouldn't it start out as "true"? isn't the component initially mounted?

Reply:
in the old implementation, `isUnmounted` is set to `false` at class construction time. this seems to be different now?

Reply:
yes, it is different now. Basically I needed a way to skip running `useIsomorphicEffect` on the first render and I reused `isUnmounted` with initial value set to `true` and then setting it to `false` inside `useEffect(()=>{ ... } , [ ])`. So `isUnmounted=false` only after first render. Perhaps `didMount` would be better name for it, but I didn't want to introduce a new variable. Also worth noting that I won't be able to use `useMounted` even if it is renamed to `didMount`.
I can't see how to make it work with the current `useMounted`, sorry. Any suggestions?